package/MDAnalysis/analysis/nucleicacids.py

```diff
@@ -140,6 +148,41 @@ def __init__(self, selection1: List[mda.AtomGroup],
             self._s1 += selection1[i]
             self._s2 += selection2[i]
 
+    @staticmethod
+    def select_strand_atoms(
+        strand1: List[Residue], strand2: List[Residue], 
+        a1_name: str, a2_name: str, guanine_name: str = 'G',
+        adenine_name: str = 'A', uracil_name: str = 'U',
+        tymine_name: str = 'T', cytosine_name: str = 'C'
+        ) -> Tuple[List[mda.AtomGroup], List[mda.AtomGroup]]:
+        pyrimidines: List[str] = [cytosine_name, tymine_name, uracil_name]
+        purines: List[str] = [adenine_name, guanine_name]
+        
+        sel1: List[mda.AtomGroup] = []
+        sel2: List[mda.AtomGroup] = []
+        strand: Tuple[Residue, Residue] = zip(strand1, strand2)
+
+
+        for s in strand:
+            if s[0].resname[0] in pyrimidines:
+                a1, a2 = a2_name, a1_name
+            elif s[0].resname[0] in purines:
+                a1, a2 = a1_name, a2_name
+            else:
+                raise ValueError(f"AtomGroup in {s} is not a valid nucleic acid")
+
+            ag1 = s[0].atoms.select_atoms(f'name {a1}')
+            ag2 = s[1].atoms.select_atoms(f'name {a2}')
+
+            if not all([len(ag) > 0 for ag in [ag1, ag2]]):
```

Comment:
the `all` function doesn't require a list, so you could reduce this to `all(len(ag) for ag in [ag1, ag2])`

Reply:
oh cool, I never knew that!